app/src/main/java/com/internship/move/presentation/authentification/viewmodel/AuthenticationViewModel.kt

```diff
@@ -15,7 +21,13 @@ class AuthenticationViewModel(
         repo.setIsUserLoggedIn(true)
     }
 
-    fun register(email: String, username: String, password: String) {
-        repo.setIsUserLoggedIn(true)
+    fun register(newUser : User) {
+        viewModelScope.launch(Dispatchers.IO) {
+            try{
+                repo.register(newUser)
+            }catch(e: Exception){
+                e.message?.let { Log.e(TAG, it) }
```

Comment:
Send the error to the ui. For instance, just print a `Something went wrong`.

It should be enough